### ISSUE TITLE
install problems

### ISSUE THREAD

**Issue by limelect** [skia4delphi/skia4delphi#39]
D10.2.3 trying to install
C:\Program Files (x86)\Embarcadero\Studio\19.0\LimLib\Skia\Packages\RAD Studio 10.2 Tokyo\Skia.Package.VCL.dproj

as a matter of fact all installations same problem

All bpl compile OK but
Directory is OK  C:\Program Files (x86)\Embarcadero\Studio\19.0\LimLib\Skia\Library\RAD Studio 10.2 Tokyo\Win32\Debug\Bpl and units too.

All files are correct and still and I get Skia.Package.RTL250.bpl  is missing
I had to move Skia.Package.RTL250.bpl to C:\Users\Public\Documents\Embarcadero\Studio\19.0\Bpl
but now I get rtl250.bpl error
In short, cannot install ver 3

**Comment by viniciusfbb** [comment]
@limelect Are you trying to install manually? If so, do you need to follow the steps in [manual installation documentation](https://github.com/skia4delphi/skia4delphi/blob/main/Documents/INSTALLATION.md#manual), especially the step 5:

> 5. Open RAD Studio, go to *Tools > Options... > IDE > Environment Variables*, and:
>    
>    5.1. Click *New...*, and add the variable name `SKIADIR`, and in *variable value* put your **Skia4Delphi** full path *(eg C:\skia4delphi)*, and click *OK*. From here, we'll call this path as `[skia4delphi_root]` in this documentation. But remember: in the step that is written `[skia4delphi_root]`, it must be filled with full path (without environment variables);
>    
>    5.2. Still in the *Environment Variables* page, in the *User System Overrides* section, find the `PATH` variable and double click to change it. You need to add the these three folders (full path) at the beginning:
>    
>    - `[skia4delphi_root]\Binary\Win32\Release;`
>      *eg C:\skia4delphi\Binary\Win32\Release;*
>    - `[skia4delphi_root]\Library\[rad_version_name]\Win32\Release\Bpl;`
>      *eg C:\skia4delphi\Library\RAD Studio 11.0 Alexandria\Win32\Release\Bpl;*
>    - `[skia4delphi_root]\Library\[rad_version_name]\Win64\Release\Bpl;`
>      *eg C:\skia4delphi\Library\RAD Studio 11.0 Alexandria\Win64\Release\Bpl;*
>    
>    5.3. Click *OK* and *Save*;

But, again I always recommend the automatic installation done by [Skia4Delphi_3.0.0_Setup.exe](https://github.com/skia4delphi/skia4delphi/releases/download/v3.0.0/Skia4Delphi_3.0.0_Setup.exe) that you will always found in [releases page](https://github.com/skia4delphi/skia4delphi/releases)

**Comment by limelect** [comment]
used exe same problem

![Screenshot - 12_02_2022 , 21_56_23](https://user-images.githubusercontent.com/4199131/153726417-9936744a-deca-4894-9eeb-650144e2f4d0.jpg)

**Comment by viniciusfbb** [comment]
The error occurs when trying to change the RAD Studio registry. Didn't you run it with the IDE open? Also see if the problem resolves when running as administrator.

**Comment by limelect** [comment]
same problem
1. closed Delphi
2. run as administratorr
3. got this

Microsoft (R) Build Engine Version 3.5.30729.5420
[Microsoft .NET Framework, Version 2.0.50727.8806]
Copyright (C) Microsoft Corporation 2007. All rights reserved.

C:\Users\LimElect\AppData\Roaming\Embarcadero\BDS\19.0\environment.proj(59,28): error MSB4024: The imported project file could not be loaded. 'Edition' is an unexpected token. The expected token is '='. Line 59, position 28.

**Comment by limelect** [comment]
Install for Delphi 11.?????
I install it for 10.2!!!!

**Comment by viniciusfbb** [comment]
> C:\Users\LimElect\AppData\Roaming\Embarcadero\BDS\19.0\environment.proj(59,28): error MSB4024: The imported project file could not be loaded. 'Edition' is an unexpected token. The expected token is '='. Line 59, position 28.

There was an error editing the environment variables of your registry. I need you to send me the values of some records:

1) Open the RegEdit (search for "regedit.exe" in windows search)
2) Access the key: HKEY_CURRENT_USER\SOFTWARE\Embarcadero\BDS\19.0\Environment Variables
3) Send me a print screen, and also paste here the content of PATH, inside this registry

**Comment by enri690** [comment]
When you click on the OK button where it reads "Please, report this issue", does it open a webpage containing more info about the error?

**Comment by viniciusfbb** [comment]
> Install for Delphi 11.
> 
> To be able to install without errors, I had to prohibit Setup to override file "enviroment.proj" in C:\Users\USERNAME\AppData\Roaming\Embarcadero\BDS\22.0\ by making it read only and after install remove read only for Bds to run.

Strangely, setup doesn't edit the enviroment.proj file, it only edits the register that RAD Studio reads when loading "enviroment.proj".

**Comment by limelect** [comment]
Yes it opens a webpage
path
$(PUBLIC)\Documents\Embarcadero\InterBase\redist\InterBase2017\IDE_spoof;$(PATH)

![Screenshot - 12_02_2022 , 22_45_46](https://user-images.githubusercontent.com/4199131/153727861-8bb418a0-51f0-42ec-a95f-93595d5b1963.jpg)

**Comment by enri690** [comment]
Setup rewrites that file every time, and then I think it is unable to parse it correctly

**Comment by enri690** [comment]
> same problem
> 
> 1. closed Delphi
> 2. run as administrator
> 3. window 7
> 4. got this
> 
> Microsoft (R) Build Engine Version 3.5.30729.5420 [Microsoft .NET Framework, Version 2.0.50727.8806] Copyright (C) Microsoft Corporation 2007. All rights reserved.
> 
> C:\Users\LimElect\AppData\Roaming\Embarcadero\BDS\19.0\environment.proj(59,28): error MSB4024: The imported project file could not be loaded. 'Edition' is an unexpected token. The expected token is '='. Line 59, position 28.

Before doing this try to make the file "enviroment.proj" Read Only

**Comment by limelect** [comment]
I did readonly
sorry same problem

**Comment by limelect** [comment]
Let's go to my original question what so complicated with install
by hand?
Never had such a problem.
something is definitely wrong with the source

**Comment by viniciusfbb** [comment]
> definitely wrong with the install source

Interesting! Apparently dcc32.exe is changing this file while compiling via command line. I'll study what may be happening, can you send me the original enviroment.proj and the one modified by setup?


> Let's go to my original question what so complicated with install by hand? Never had such a problem. something is definitely wrong with the install source
There's no problem with the manual installation, it's just that it's common to skip some steps, especially step 5. The documentation I sent you describes the step by step very well.

**Comment by limelect** [comment]
where is it? "The documentation I sent you describes the step by step very well."?

**Comment by jatinpandya** [comment]
I have the same issue - the installer shows message to report the error - someone has already uploaded the screen grab above. And then I am shown the following BuildLogs.txt. Is there a way to avoid this error? I have already tried installing without IDE running, as administrator, etc.

`Microsoft (R) Build Engine version 4.8.4161.0
[Microsoft .NET Framework, version 4.0.30319.42000]
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 2/14/2022 9:02:21 AM.
Project "C:\Work\Skia4Delphi\Packages\RAD Studio 11.0 Alexandria\Skia.Package.RTL.dproj" on node 1 (default targets).
C:\Users\Jatin\AppData\Roaming\Embarcadero\BDS\22.0\environment.proj(7,9): error MSB5016: The name "asl.log" contains an invalid character ".". [C:\Work\Skia4Delphi\Packages\RAD Studio 11.0 Alexandria\Skia.Package.RTL.dproj]
Done Building Project "C:\Work\Skia4Delphi\Packages\RAD Studio 11.0 Alexandria\Skia.Package.RTL.dproj" (default targets) -- FAILED.

Build FAILED.

"C:\Work\Skia4Delphi\Packages\RAD Studio 11.0 Alexandria\Skia.Package.RTL.dproj" (default target) (1) ->
  C:\Users\Jatin\AppData\Roaming\Embarcadero\BDS\22.0\environment.proj(7,9): error MSB5016: The name "asl.log" contains an invalid character ".". [C:\Work\Skia4Delphi\Packages\RAD Studio 11.0 Alexandria\Skia.Package.RTL.dproj]

    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:00.02
`

**Comment by viniciusfbb** [comment]
We are doing some tests and will release a new version by tomorrow with some corrections and with the generation of logs by the setup to facilitate the analysis of this type of problem.

For now, manual installation is the alternative. You will find the documentation for manual installation here: https://github.com/skia4delphi/skia4delphi/blob/main/Documents/INSTALLATION.md#manual

**Comment by enri690** [comment]
> I have the same issue - the installer shows message to report the error - someone has already uploaded the screen grab above. And then I am shown the following BuildLogs.txt. Is there a way to avoid this error? I have already tried installing without IDE running, as administrator, etc.
> 
> `Microsoft (R) Build Engine version 4.8.4161.0 [Microsoft .NET Framework, version 4.0.30319.42000] Copyright (C) Microsoft Corporation. All rights reserved.
> 
> Build started 2/14/2022 9:02:21 AM. Project "C:\Work\Skia4Delphi\Packages\RAD Studio 11.0 Alexandria\Skia.Package.RTL.dproj" on node 1 (default targets). C:\Users\Jatin\AppData\Roaming\Embarcadero\BDS\22.0\environment.proj(7,9): error MSB5016: The name "asl.log" contains an invalid character ".". [C:\Work\Skia4Delphi\Packages\RAD Studio 11.0 Alexandria\Skia.Package.RTL.dproj] Done Building Project "C:\Work\Skia4Delphi\Packages\RAD Studio 11.0 Alexandria\Skia.Package.RTL.dproj" (default targets) -- FAILED.
> 
> Build FAILED.
> 
> "C:\Work\Skia4Delphi\Packages\RAD Studio 11.0 Alexandria\Skia.Package.RTL.dproj" (default target) (1) -> C:\Users\Jatin\AppData\Roaming\Embarcadero\BDS\22.0\environment.proj(7,9): error MSB5016: The name "asl.log" contains an invalid character ".". [C:\Work\Skia4Delphi\Packages\RAD Studio 11.0 Alexandria\Skia.Package.RTL.dproj]
> 
> ```
> 0 Warning(s)
> 1 Error(s)
> ```
> 
> Time Elapsed 00:00:00.02 `

1. Remove the entire line that contain asl.log in your enviroment.proj,
2. Save the file but be sure it does not take the extension txt.
3. Make file Read Only.
4. Install
5. Remove attribute Read Only.

**Comment by HBenzaoui** [comment]
> Install for Delphi 11.
> 
> To be able to install without errors, I had to prohibit Setup to override file "enviroment.proj" in C:\Users\USERNAME\AppData\Roaming\Embarcadero\BDS\22.0\ by making it read only and after install remove read only for Bds to run.

That worked for me !thanks

**Comment by lidernetworktecnologia** [comment]
> > Install for Delphi 11.
> > To be able to install without errors, I had to prohibit Setup to override file "enviroment.proj" in C:\Users\USERNAME\AppData\Roaming\Embarcadero\BDS\22.0\ by making it read only and after install remove read only for Bds to run.
> 
> That worked for me !thanks

Fiz esse procedimento e o instalador funcionou para Rio e Sidney

**Comment by viniciusfbb** [comment]
First, I apologize for the installation inconvenience.

Right now, we've released a new release with several fixes/improvements included in the setup.

Hopefully this will resolve most installation issues in this new release. I would appreciate it if someone with problems in the previous version would give feedback on this new one.

**Comment by enri690** [comment]
Install for Alexandria goes well. Demo Android runs Ok also. Thank you.
